### DISCUSSION TITLE
Is nested link not supported?

### DISCUSSION THREAD

**Discussion by ratson** [geldata/gel#4033] [Q&A] [answered]
edgedb version:  `1.4+5ffe34`

```
module default {
  type Link {
    multi link links -> Link {
      link another -> Link;
    }
  }
}
```

`edgedb migration create` gives the following error,

```
error: Unexpected 'another'
  ┌─ ./dbschema/default.esdl:4:12
  │
4 │       link another -> Link;
  │            ^^^^^^^ error

edgedb error: cannot proceed until .esdl files are fixed
```

The error is not explaining why it is failing.

Is this not a supported syntax?

**Comment by vpetrovykh** [comment]
You can only have _properties_ attached to links (https://www.edgedb.com/docs/datamodel/links#link-properties). Links themselves have to be part of object types.

What are you trying to achieve with this "nested link" structure?

One way to define a nested structure is:
```
module default {
  type Link {
    multi link links -> Link;
    link another -> Link;
  }
}
```
Then you can query the nested structure like so:
```
select Link {
  links: {
    another
  }
}
```

Potentially you may want 2 different types, though:
```
module default {
  type Link0 {
    multi link links -> Link1;
  }

  type Link1 {
    link another -> Link0;
  }
}
```

**Comment by ratson** [comment, marked as answer]
@vpetrovykh Thank you for the answer.

The second solution is close to what I am looking for, here is my modified version.

```
module default {
  type Link {
    multi link links -> NestedLink;
  }

  type NestedLink extending Link {
     link another -> Link;
  }
}
```

Not optimal, but good enough.
